src/main/java/me/moodcat/api/SongAPI.java

```diff
@@ -48,4 +62,100 @@ public Song getSongById(@PathParam("id") final int id) {
         return songDAO.findById(id);
     }
 
+    /**
+     * Process a user classification for the given songId.
+     *
+     * @param id
+     *            The id of the song.
+     * @param classification
+     *            The classification of the user for the song.
+     * @return The classification provided.
+     */
+    @POST
+    @Path("{id}/classify")
+    @Transactional
+    public ClassificationRequest classifySong(@PathParam("id") final int id,
+            final ClassificationRequest classification) {
+        final Song song = this.songDAO.findById(id);
+
+        if (song.getNumberOfPositiveVotes() < MINIMUM_NUMBER_OF_POSITIVE_VOTES) {
+            final VAVector classificationVector = new VAVector(
+
+                    classification.getValence(),
+                    classification.getArousal());
```

Comment:
It's probably a good idea to add a check whether the valence and arousal have valid values here, and ignore the vector if it's not valid. Else either the asserts of the VAVector will fail (and throw an exception), or any non-valid values will be used.